src/soroban/server.ts

```diff
@@ -235,6 +235,62 @@ export class Server {
     );
   }
 
+  /**
+   * Retrieves the WASM bytecode for a given contract.
+   *
+   * This method allows you to fetch the WASM bytecode associated with a contract
+   * deployed on the Soroban network. The WASM bytecode represents the executable
+   * code of the contract.
+   *
+   * @param {string} contractId    the contract ID containing the
+   *    WASM bytecode to retrieve
+   *
+   * @returns {Promise<Buffer>}   a Buffer containing the WASM bytecode
+   *
+   * @throws {Error} If the contract or its associated WASM bytecode cannot be
+   *    found on the network.
+   *
+   * @example
+   * const contractId = "CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5";
+   * server.getContractWasm(contractId).then(wasmBuffer => {
+   *   console.log("WASM bytecode length:", wasmBuffer.length);
+   *   // ... do something with the WASM bytecode ...
+   * }).catch(err => {
+   *   console.error("Error fetching WASM bytecode:", err);
+   * });
+   */
+  public async getContractWasm(
+    contractId: string
+  ): Promise<Buffer> {
+    const contractLedgerKey = new Contract(contractId).getFootprint();
+    const response = await this.getLedgerEntries(contractLedgerKey);
+    if (!response.entries.length || !response.entries[0]?.val) {
+      return Promise.reject({code: 404, message: `Could not obtain contract hash from server`});
```

Comment:
throwing is fine here!

Reply:
I changed it back to Promise.reject since I was following the pattern in the get_contract_data.js test which checks for a 404 code. I can revert it to throwing an error if you'd prefer.